### DISCUSSION TITLE
[Dev] Is there a better way than carry the binary and lib for `GULP` in the source code and package?

### DISCUSSION THREAD

**Discussion by DanielYang59** [materialsproject/pymatgen#3788] [General]
Solved in #3789.

Hope this is not a silly question. Just wondering why do we need to carry the binary and libs for `GULP` (`pymatgen/cmd_line/gulp`) and also include it inside the pypi package?
https://github.com/materialsproject/pymatgen/blob/ff94faa5bb490354c420f188f1abeac17cb40931/setup.py#L112

This doesn't sound like part of our `pymatgen` package (just like we don't ship other external dependencies like `Bader` or such)?

**Comment by janosh** [comment]
no idea. this predates my involvement with `pymatgen`. i suspect there's no (longer?) good reason to package `gulp`. maybe @shyuep can comment?

**Reply by shyuep** [reply]
Gulp caller uses gulp.

**Reply by DanielYang59** [reply]
Can we locate this out of our package? Just like `bader/chargemol/**__caller` uses some external packages, we cannot just include all these packages inside our package?

**Reply by janosh** [reply]
what's the reason for forcing every user to download an 11 year old Linux 64-bit only gulp binary? i'm guessing the intersection of gulp and active pymatgen users is close to zero

**Reply by DanielYang59** [reply]
Yes it's pretty old and I highly doubt if we really need to pack it inside `pymatgen` package. 

Also I cannot access the website for some reason: https://github.com/materialsproject/pymatgen/blob/ff94faa5bb490354c420f188f1abeac17cb40931/pymatgen/command_line/gulp_caller.py#L1-L4

But this website from Curtin University seems to work: https://gulp.curtin.edu.au/index.html
